osquery/logger/plugins/kafka_producer.cpp

```diff
@@ -26,17 +23,44 @@
 namespace osquery {
 
 FLAG(string,
-     logger_kafka_brokers,
+     kafka_brokers,
      "localhost",
      "Bootstrap broker(s) as a comma-separated list of host or host:port "
      "(default port 9092)");
+FLAG_ALIAS(std::string, logger_kafka_brokers, kafka_brokers);
 
-FLAG(string, logger_kafka_topic, "", "Kafka topic to publish logs under");
+FLAG(string, kafka_topic, "osquery", "Kafka topic to publish logs under");
+FLAG_ALIAS(std::string, logger_kafka_topic, kafka_topic);
 
 FLAG(string,
-     logger_kafka_acks,
+     kafka_acks,
      "all",
      "The number of acknowledgments the leader has to receive (0, 1, 'all')");
+FLAG_ALIAS(std::string, logger_kafka_acks, kafka_acks);
+
+FLAG(string,
+     kafka_security_protocol,
+     "plaintext",
+     "Protocol used to communicate with brokers ('plaintext', 'ssl')");
+
+FLAG(string, kafka_ssl_cipher_suites, "", "List of cipher suites");
+
+FLAG(string,
+     kafka_ssl_key_location,
+     "",
+     "Path to client's private key (PEM) used for authentication");
+
+FLAG(string, kafka_ssl_key_password, "", "Private key passphrase");
+
+FLAG(string,
+     kafka_ssl_certificate_location,
```

Comment:
You also have [`tls_client_cert`](https://github.com/facebook/osquery/blob/master/osquery/remote/transports/tls.cpp#L47) and `tls_client_key` if you want to overload their use. I'd highly recommend using `tls_server_certs` to select the CA/intermediate/etc, but for the client cert+key, Kafka may want to differentiate between the optional pair used for the the TLS plugins.

Reply:
I know we've set bad precedence, but given the above comment, if this flag does remain, can it be `kafka_tls_client_cert`?